job-dsls/jobs/tools_provisioning_verdaccio.groovy

```diff
@@ -0,0 +1,201 @@
+/**
+ * Creates provision-jobs in Jenkins/Tools/provisioning for
+ * provision-verdaccio-service
+ */
+
+def final DEFAULTS = [
+        folderPath : "Provisioning",
+        logRot : 10,
+        labExp : "ansible",
+        params : [ ],
+        openJdk : "openjdk1.8_local"
+]
+
+def final JOB_NAMES = [
+        "provision-verdaccio-service"  : [
+                params : [
+                        [name: "IMAGE", default: "rhel-7.6-server-x86_64-released", description: "The name of the image to be used for machine creation." ],
+                        [name: "FLAVOUR", default: "ci.m1.medium.no.nested.virt", description: "The flavor (i.e. resources such as CPU cores, RAM, ...) defining the machine. m1.medium = 2 vCPUs, 4 GB RAM, 40 GB HDD" ]
+                ],
+                jobDescription : "The job for provisioning verdaccio service node."
+        ]
+]
+
+//create folders
+folder("Provisioning")
+
+for (jobNames in JOB_NAMES) {
+    Closure<Object> get = { String key -> jobNames.value[key] ?: DEFAULTS[key] }
+
+    String jobName = jobNames.key
+    String folderPath = get("folderPath")
+    String labExp = get("labExp")
+    String openJdk = get("openJdk")
+    String jobDescription = get("jobDescription")
+    def params = get("params")
+    def logRot = get("logRot")
+
+    // jobs for master branch don't use the branch in the name
+    String jobN = "$folderPath/$jobName"
+
+    job(jobN) {
+        description(jobDescription)
+
+        logRotator {
+            numToKeep(logRot)
+        }
+
+        label(labExp)
+
+        jdk(openJdk)
+
+        for ( param in params ) {
+            parameters {
+                stringParam(param.name, param.default, param.description)
+                password{
+                    name('ENCRYPTED_PASSWORD')
+                    defaultValue('redhat')
+                    description('The encrypted password. To generate it you should take value from npm adduser command. Default redhat')
+                }
+            }
+        }
+
+        // Allows a job to check out sources from an SCM provider.
+        scm {
+            git {
+                remote {
+                    url('git@github.com:kiegroup/kie-jenkins-scripts.git')
+                    credentials("kie-qe-ci-user-key")
+                    branch('*/BXMSPROD-1147')
```

Comment:
@mbiarnes is this branch correct?

Reply:
@Ginxo ^^

Reply:
maybe we can use standard branch or a tag?

Reply:
it should be master